src/Streamly/Internal/Data/Fold.hs

```diff
@@ -818,9 +827,9 @@ drainN n = ltake n drain
 
 -- | A fold that drains elements of its input as long as the predicate succeeds,
 -- running the effects and discarding the results.
-{-# INLINABLE drainSepBy #-}
-drainSepBy :: Monad m => (a -> Bool) -> Fold m a ()
-drainSepBy p = takeSepBy p drain
+{-# INLINABLE drainWhile #-}
```

Comment:
`drainWhile`'s behaviour does not justify its name. An extra element is drained.

Reply:
Moved to parsers.